src/lib.rs

```diff
@@ -2020,13 +2020,10 @@ macro_rules! transmute_ref {
 
             // `u` is inferred to have type `U` because it's used as `&u` as the
             // value returned from this branch.
-            //
-            // SAFETY: This code is never run.
-            let u = unsafe {
-                 // Clippy: It's okay to transmute a type to itself.
-                #[allow(clippy::useless_transmute)]
-                $crate::macro_util::core_reexport::mem::transmute(t)
-            };
+            let u;
+
+            $crate::assert_size_eq!(t, u);
+
             &u
```

Comment:
Slight stylistic preference for grouping these together, but feel free to ignore this.

Same in the next branch.

```suggestion
            let u;
            $crate::assert_size_eq!(t, u);
            &u
```